src/Neo/SmartContract/Native/NativeContract.cs

```diff
@@ -160,6 +165,48 @@ protected NativeContract()
             contractsDictionary.Add(Hash, this);
         }
 
+        /// <summary>
+        /// It is the initialize block
+        /// </summary>
+        /// <param name="settings">The <see cref="ProtocolSettings"/> where the HardForks are configured.</param>
+        /// <param name="index">Block index</param>
+        /// <returns>True if the native contract must be initialized</returns>
+        internal bool IsInitializeBlock(ProtocolSettings settings, uint index)
+        {
+            if (ActiveIn != Hardfork.HF_Genesis)
+            {
+                if (!settings.Hardforks.TryGetValue(ActiveIn, out var activeIn))
+                {
+                    throw new InvalidOperationException($"The hardfork '{ActiveIn}' required for {Name} is not configured in ProtocolSettings.");
```

Comment:
We have a logic that allows (for example) to configure `Basilisk: 5` in the node configuration without `Aspidochelone` being explicitly specified. This means that `Aspidochelone` is treated to be enabled since the genesis. And here if, for example, CryptoLib should be enabled in `Aspidochelone`, we will throw an exception, because `Aspidochelone` is not in the node's config.

We should align this part of code wrt [`IsHardforkEnabled` behaviour](https://github.com/neo-project/neo/blob/efa86592d080ebfde685bc87a26177c5bed5431f/src/Neo/SmartContract/ApplicationEngine.cs#L614), e.g. to retrieve the enabling height for hardfork even if it's not configured, and then compare `index` with this height. Or we can slightly change the hardfork initialization scheme to directly specify zero height for all hardforks that preceed the configured one, check how it is done in NeoGo: https://github.com/nspcc-dev/neo-go/blob/master/pkg/core/blockchain.go#L297-L305 (and of course, all hardforks [pass validation](https://github.com/nspcc-dev/neo-go/blob/c5e284c117a997656963ecfb27b9ee470a7bdc45/pkg/config/protocol_config.go#L99-L116) before this explicit initialization according to rules described in https://github.com/neo-project/neo/pull/2886).

Reply:
It is not about whether it is safe for us to do so, this is not only about few of us who is maintaining the core, but the whole community who are also developing their tools. If we just change existing public data types at well, things will be problematic for others to follow.

Reply:
The existing public data are not changed in the highlighted part of code, the comment is about other problem.

Reply:
Fixed